spec/models/manageiq/providers/vmware/infra_manager/inventory/collector_spec.rb

```diff
@@ -535,6 +561,27 @@ def dvpg_rename_object_update
       )
     end
 
+    def vm_add_new_custom_value_update
+      RbVmomi::VIM.ObjectUpdate(
+        :kind       => "modify",
+        :obj        => RbVmomi::VIM.VirtualMachine(vim, "vm-107"),
+        :changeSet  => [
+          RbVmomi::VIM.PropertyChange(:name => "availableField", :op => "assign", :val => [RbVmomi::VIM.CustomFieldDef(:key => 300, :managedObjectType => "VirtualMachine", :name => "foo", :type => "string")]),
+          RbVmomi::VIM.PropertyChange(:name => "summary.customValue[300]", :op => "add", :val => RbVmomi::VIM.CustomFieldStringValue(:key => 300, :value => "bar"))
+        ]
+      )
+    end
+
+    def vm_edit_custom_value_update
+      RbVmomi::VIM.ObjectUpdate(
+        :kind       => "modify",
+        :obj        => RbVmomi::VIM.VirtualMachine(vim, "vm-107"),
+        :changeSet  => [
+          RbVmomi::VIM.PropertyChange(:name => "summary.customValue[300]", :op => "assign", :val => RbVmomi::VIM.CustomFieldStringValue(:key => 300, :value => "baz"))
```

Comment:
Editing an existing custom attribute (aka array assign vs. array add) was triggering the "Array properties not supported yet" error, added a test case for this as well